huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_parameter_template.go

```diff
@@ -55,21 +55,41 @@ func ResourceGaussDBMysqlTemplate() *schema.Resource {
 				Description: `Specifies the parameter template description.`,
 			},
 			"datastore_engine": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Computed:    true,
-				ForceNew:    true,
-				Description: `Specifies the DB engine.`,
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				RequiredWith:  []string{"datastore_version"},
+				ConflictsWith: []string{"source_configuration_id", "instance_id"},
+				Description:   `Specifies the DB engine.`,
 			},
 			"datastore_version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"datastore_engine"},
+				Description:  `Specifies the DB version.`,
+			},
+			"source_configuration_id": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Computed:    true,
 				ForceNew:    true,
-				Description: `Specifies the DB version.`,
-				RequiredWith: []string{
-					"datastore_engine",
-				},
+				Description: `Specifies the parameter template name.`,
+			},
+			"instance_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"instance_configuration_id"},
+				Description:  `Specifies the parameter template name.`,
+			},
```

Comment:
```suggestion
			"instance_id": {
				Type:         schema.TypeString,
				Optional:     true,
				ForceNew:     true,
				RequiredWith: []string{"instance_configuration_id"},
				ConflictsWith: []string{"source_configuration_id", "datastore_engine"},
				Description:  `Specifies the parameter template name.`,
			},
```

Reply:
it has been added to the param of datastore_engine